risc0/circuit/rv32im/src/prove/emu/preflight/mod.rs

```diff
@@ -872,31 +872,33 @@ impl<'a> BigInt2Witness<'a> {
 }
 
 impl<'a> bibc::BigIntIO for BigInt2Witness<'a> {
-    fn load(&mut self, arena: u32, offset: u32, count: u32) -> Result<BigUint> {
+    fn load(&mut self, arena: u32, offset: u32, count: u32) -> Result<UBig> {
         // tracing::debug!("load(arena: {arena}, offset: {offset}, count: {count})");
         let base = ByteAddr(self.preflight.peek_register(arena as usize)?);
         let addr = base + offset * BIGINT2_WIDTH_BYTES as u32;
         let bytes = self
             .preflight
             .peek_region(addr.waddr(), count / WORD_SIZE as u32)?;
-        Ok(BigUint::from_bytes_le(&bytes))
+        Ok(UBig::from_le_bytes(&bytes))
     }
 
-    fn store(&mut self, arena: u32, offset: u32, count: u32, value: &BigUint) -> Result<()> {
+    fn store(&mut self, arena: u32, offset: u32, count: u32, value: &UBig) -> Result<()> {
         let base = ByteAddr(self.preflight.peek_register(arena as usize)?);
         let addr = base + offset * BIGINT2_WIDTH_BYTES as u32;
         // tracing::debug!(
         //     "store(arena: {arena}, offset: {offset}, count: {count}, value: {value}, base: {base:?}, addr: {addr:?})"
         // );
         let addr = addr.waddr();
-        let mut words = value.to_u32_digits();
+        let bytes = value.to_le_bytes();
         let word_count = count as usize / WORD_SIZE;
-        if words.len() < word_count {
-            words.resize(word_count, 0);
-        }
-        ensure!(words.len() == word_count);
-        for (i, word) in words.iter().enumerate() {
-            self.witness.insert(addr + i, *word);
+        for i in 0..word_count {
+            let offset = i * 4;
+            let mut word = 0_u32;
+            if offset < bytes.len() {
+                let wordbytes = &bytes[offset..offset + 4];
+                word = u32::from_le_bytes(wordbytes.try_into().unwrap());
```

Comment:
Hmm. I would think this could fail if storing something very small (e.g. `1`) as then `bytes` would have a length that wasn't an even multiple of 4 and so this line would try to build `word` from fewer than 4 bytes and fail. But we have tests that should catch this in https://github.com/risc0/risc0/blob/main/risc0/bigint2/src/field/tests.rs ... and I'm not sure those tests are being run. Not this PR's fault if they aren't, but I'm not going to approve until I investigate that more

Reply:
Update: Yep, this causes those tests to fail, for the reason I outlined. I'll make a separate update with the tests working.

Reply:
https://github.com/risc0/risc0/pull/2763

Reply:
Good catch! I had assumed that a bigint with 32-bit limbs would always return a multiple-of-4 byte array.

Reply:
Also, we need to avoid calling `unwrap` when possible and return errors instead.